# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1265697682 -3600
#      Tue Feb 09 07:41:22 2010 +0100
# Node ID 30078b201a2c9e5e630baeeb7b6a962598533c98
# Parent  b122dd3075ce6331331c794fee092f22d7467ab9
replace eval by sscanf in menu.m

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-02-09  Jaroslav Hajek  <highegg@gmail.com>
+
+	* miscellaneous/menu.m: Use sscanf rather than eval.
+
 2010-02-09  Jaroslav Hajek  <highegg@gmail.com>
 
 	* miscellaneous/substruct.m: Vectorize.
 
 2010-02-08  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/accumarray.m: Document complexity.
 
diff --git a/scripts/miscellaneous/menu.m b/scripts/miscellaneous/menu.m
--- a/scripts/miscellaneous/menu.m
+++ b/scripts/miscellaneous/menu.m
@@ -58,17 +58,17 @@ function num = menu (t, varargin)
 
     while (1)
       for i = 1:nopt
         printf ("  [%2d] ", i);
         disp (varargin{i});
       endfor
       printf ("\n");
       s = input ("pick a number, any number: ", "s");
-      eval (sprintf ("num = %s;", s), "num = [];");
+      num = sscanf (s, "%d");
       if (! isscalar (num) || num < 1 || num > nopt)
         printf ("\nerror: input invalid or out of range\n\n");
       else
         break;
       endif
     endwhile
 
   unwind_protect_cleanup
